app/components/certificate/certificate-request.tsx

```diff
@@ -38,9 +41,16 @@ export default function CertificateRequestView({ domain, onRequest }: Certificat
           <Text>{domain}</Text>
         </Flex>
       </Flex>
-      <Button width={{ base: 'full', md: '3xs' }} shadow="xl" onClick={onRequest}>
-        Request a Certificate
-      </Button>
+      <Form method="post" onSubmit={() => setIsDisabled(true)}>
+        <Button
+          type="submit"
+          width={{ base: 'full', md: '3xs' }}
+          shadow="xl"
+          isDisabled={isDisabled}
```

Comment:
This is good for now, but I wonder if in a follow-up we should update the page with some text that says, "OK, we're doing it, hang tight..."